app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -239,6 +243,25 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                         .zoom(ZOOM_LEVEL)
                         .build();
                 mapBoxMap.setCameraPosition(cameraPosition);
+                scaleBarPlugin = new ScaleBarPlugin(mapView, mapBoxMap);
+
+                int color = isDarkTheme?R.color.bottom_bar_light:R.color.bottom_bar_dark;
+
+                float size = new TextView(getContext()).getTextSize()-1;
+
+                scaleBarOptions = new ScaleBarOptions(getContext())
+                    .setTextColor(color)
+                    .setTextSize(size)
+                    .setBarHeight((float) (size/3.5))
+                    .setBorderWidth(size/10)
+                    .setMetricUnit(true)
+                    .setRefreshInterval(15)
```

Comment:
Can you use constants in place of these [_magic numbers_](https://stackoverflow.com/a/47902/5614968)? Some comments about _why_ these numbers were chosen would be nice.

Reply:
i can do that but only after view is created(due to line 250 above). So basically code will be like this:

final float SIZE = new TextView(getContext()).getTextSize()-1;
                //Default text size for scalebar params
                final float SCALEBAR_HEIGHT = (float) (SIZE/3.5);
                final float SCALEBAR_BORDER_WIDTH = SIZE/10;
                final float SCALEBAR_MARGIN = SIZE/2;
                //these values are choosen to keep scalebar size comparable to other text depending upon the screen size and device resolution
                final int REFRESH_INTERVAL = 15;
                //minimum refresh time in millisec

                scaleBarOptions = new ScaleBarOptions(getContext())
                    .setTextColor(color)
                    .setTextSize(SIZE)
                    .setBarHeight(SCALEBAR_HEIGHT)
                    .setBorderWidth(SCALEBAR_BORDER_WIDTH)
                    .setMetricUnit(true)
                    .setRefreshInterval(REFRESH_INTERVAL)
                    .setMarginTop(SCALEBAR_MARGIN)
                    .setMarginLeft(SCALEBAR_MARGIN)
                    .setTextBarMargin(SCALEBAR_MARGIN);

Should i make these changes?

Reply:
As we don't use these numbers again anywhere in the class, you can just declare them as `final` variables local to that method. No need to declare them as instance variables.

Also, you seem to be positioning the comments below the corresponding line. It's customary to actually have the related comments above the line or at the end of it.

The content of the comments look nice. It would be if could also add the reason for why you're choosing `3.5` to get the scalebar height, if any.

Reply:
okay will do that👍